### ISSUE TITLE
Markdown links containing template tags aren't treated as links

### ISSUE THREAD

**Issue by ghost** [tbroadley/spellchecker-cli#123]
This Markdown:

```md
[something](https://example.com/my-version/{{ jinja2_var }}/this-is-something)
```

Causes spellchecker-cli to complain that portions of the URL are misspelt:

```
something.md
    1:1-12 warning `my-version` is misspelt; did you mean?
```

This is using in mkdocs with the mkdocs-macros plugin to parameterize (single-source) some content.

Is it possible tell it to ignore those URL?

**Comment by tbroadley** [comment]
Thank you for the issue!

To work around this, I'd suggest adding `my-version` to a personal dictionary and configuring Spellchecker to use it: https://github.com/tbroadley/spellchecker-cli?tab=readme-ov-file#personal-dictionaries

Another workaround: A couple of online Markdown say that

```md
[something](https://example.com/my-version/{{jinja2_var}}/this-is-something)
```

Is a valid Markdown link. You could consider removing the spaces in `{{ jinja2_var }}` to get Spellchecker to treat the link as a link and ignore `my-version`.

---

In terms of actually fixing the bug: I expect the problem is [remark](https://github.com/remarkjs/remark) (the Markdown parser used by this tool) isn't parsing the link as a link because of the `{{ jinja2_var }}` inside it. Off the top of my head, I don't see an easy way for the tool to identify cases like this. I wouldn't expect remark to parse this as a link. It isn't a valid Markdown link, until mkdocs-macros replaces `{{ jinja2_var }}` with something else.

That gives me another idea. You could consider running Spellchecker on the output of mkdocs. However, mkdocs outputs HTML, right? This tool doesn't support spellchecking HTML right now. But it could. I'm imagining it would use [rehype-retext](https://github.com/rehypejs/rehype-retext) to translate HTML into text that can be spellchecked, just like Spellchecker uses remark-retext for Markdown right now.

To summarize, I think the current behaviour is reasonable. It seems good for Spellchecker to support spellchecking HTML. I've just filed https://github.com/tbroadley/spellchecker-cli/issues/124 to track that.
